lambda/lambda.mli

```diff
@@ -762,9 +762,31 @@ and lambda_event_kind =
   | Lev_function
   | Lev_pseudo
 
+(* Descriptor for a parameter that this module takes at runtime. *)
+type runtime_param_descr =
+  | Rp_argument_block of Global_module.t  (* The argument block of a module
+                                             compiled with [-as-argument-for] *)
+  | Rp_dependency of Global_module.t      (* A parameterised module (not itself a
+                                             parameter) that this module depends
+                                             on *)
+  | Rp_unit                               (* The unit value (only used when
+                                             there are no other parameters) *)
+
+type module_block_format =
+  | Mb_record of { mb_size : int }      (* A block with [mb_size] fields *)
+  | Mb_wrapped_function of { mb_runtime_params : runtime_param_descr list;
+                             mb_returned_size : int;
+                           }
+                                        (* A block with exactly one field:
+                                           a function taking [mb_runtime_params] and
+                                           returning a block with
+                                           [mb_returned_size] fields *)
+
 type program =
   { compilation_unit : Compilation_unit.t;
-    main_module_block_size : int;
+    module_block_format : module_block_format;
+    arg_block_field : int option;       (* Unnamed field with argument block
```

Comment:
I've renamed `module_block_format` to `main_module_block_format` since it wasn't clear that it only describes the main module block. `arg_block_field` describes where we put the argument block, which indeed only exists when someone passes `-as-argument-for`

Reply:
If it's not essential that the argument field is itself a block, I suggest we just call it "argument_field".

Reply:
Its block-ness isn't essential here, but the _concept_ of “argument block” is. (I think you suggested elsewhere calling it `field_idx` so I did add the `_idx`.)

Reply:
My complaints against mentioning block here is that there is already main module block. So, is argument block parallel to the main module block? No, it's in fact a block inside a block. I think we should at least have comments saying it's the index INSIDE the main module block.

Reply:
Ah, it's a sub-block of the main module block. That could certainly stand to be clearer. Adding comments.